luup files/L_RGBController1.lua

```diff
@@ -1037,6 +1037,49 @@ function doColorTransition(lul_device)
 	end
 end
 
+-- MySensor RGB(W)
+RGBDeviceTypes["MYS-RGBW"] = {
+	getParameters = function (lul_device)
+		return {
+			name = "MySensors RGBW",
+			settings = {
+				{ variable = "ArduinoId", name = "Arduino plugin Id", type = "string" },
+				{ variable = "RadioId",   name = "RGB Node altid",    type = "string" }
+			}
+		}
+	end,
+
+	getColorChannelNames = function (lul_device)
+		return {"red", "green", "blue", "warmWhite", "coolWhite"}
```

Comment:
Hi vosmont,

You'r right, about exposing all actions implementing the action from
I_RGBController1.xml
into I_Arduino1.xml. I ever tried to do so but then still no luck.
For some reason the RGB-controller initialization does not complete.
I also though about your dummy RGB device idea, this is actually a bit what
i propose, In the MySensors forum, i.s.o. the dummy devices  i suggest to
remove the MySensors created one.

Why would control several RGB modules with one RGB controller instance?
You could create multiple instances of the RGBcontroller one for each
device.

Greetz, Bart

On Wed, Oct 14, 2015 at 6:02 PM, vosmont <notifications@github.com> wrote:

> In luup files/L_RGBController1.lua
> <https://github.com/vosmont/Vera-Plugin-RGBController/pull/1#discussion_r42013623>
> :
>
> > @@ -1037,6 +1037,49 @@ function doColorTransition(lul_device)
> >  	end
> >  end
> >
> > +-- MySensor RGB(W)
> > +RGBDeviceTypes["MYS-RGBW"] = {
> > +	getParameters = function (lul_device)
> > +		return {
> > +			name = "MySensors RGBW",
> > +			settings = {
> > +				{ variable = "ArduinoId", name = "Arduino plugin Id", type = "string" },
> > +				{ variable = "RadioId",   name = "RGB Node altid",    type = "string" }
> > +			}
> > +		}
> > +	end,
> > +
> > +	getColorChannelNames = function (lul_device)
> > +		return {"red", "green", "blue", "warmWhite", "coolWhite"}
>
> I looked more in detail at your need. There's a problem in the way you use
> the RGB Controller.
>
> What I understood : the plugin MySensors creates a RGB device with reusing
> service ""urn:schemas-upnp-org:device:RGBController:1" and device
> definition "D_RGBController1.xml".
>
> It's just possible if the plugin MySensors exposes all the actions
> implemented in "I_RGBController1.xml". Besides, I have to updated the RGB
> Controller plugin because it was not designed to keep the params of several
> devices.
>
> Another approach would be to create a dummy RGB device with a specific
> service for arduino RGB, and then control it by the current way of the RGB
> Controller plugin (using the code you have added).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vosmont/Vera-Plugin-RGBController/pull/1/files#r42013623>
> .
>